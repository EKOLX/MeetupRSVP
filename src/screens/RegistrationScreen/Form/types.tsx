import { InputsModel } from "../types";

export interface Props {
  values: InputsModel;
  errors: Partial<InputsModel>;
  onInputChange: (text: string, input: keyof InputsModel) => void;
  onError: (errorMessage: string | null, input: keyof InputsModel) => void;
  onSubmit: () => void;
}
