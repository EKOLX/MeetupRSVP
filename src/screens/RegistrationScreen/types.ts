export interface InputsModel {
  name: string;
  age: string;
  birthdate: string;
  profession: string;
  locality: string;
  guestsCount: string;
  address: string;
}

export interface Validation {
  valid: boolean;
  inputs: InputsModel;
}
