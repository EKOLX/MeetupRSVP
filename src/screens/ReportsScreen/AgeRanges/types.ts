import { AgeRange } from "../types";

export interface Props {
  ageRanges: Array<AgeRange>;
}
