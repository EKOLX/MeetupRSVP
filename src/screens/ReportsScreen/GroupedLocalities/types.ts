import { Locality } from "../types";

export interface Props {
  localities: Array<Locality>;
}
