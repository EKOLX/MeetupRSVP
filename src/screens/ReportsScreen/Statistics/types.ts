export interface Props {
  averageGroupSize: number;
  employedCount: number;
  studentCount: number;
}
