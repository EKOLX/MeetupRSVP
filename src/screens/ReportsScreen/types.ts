export interface AgeRange {
  ageRange: string;
  count: number;
}

export interface Locality {
  locality: string;
  count: 0;
}
