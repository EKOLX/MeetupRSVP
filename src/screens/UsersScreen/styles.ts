import { StyleSheet } from "react-native";

import Colors from "../../constants/Colors";

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    backgroundColor: Colors.white,
    padding: 32,
  },
  searchInput: {
    width: "100%",
    borderWidth: 1,
    borderRadius: 6,
    borderColor: Colors.gray,
    padding: 6,
  },
  userItem: {
    flexDirection: "row",
    justifyContent: "space-between",
    borderBottomWidth: 1,
    borderBottomColor: Colors.violet,
    paddingVertical: 6,
  },
  usersContainer: {
    width: "100%",
    marginTop: 22,
  },
});
