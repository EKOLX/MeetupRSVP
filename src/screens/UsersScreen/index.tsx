import { useEffect, useState } from "react";
import {
  FlatList,
  ListRenderItem,
  Pressable,
  Text,
  TextInput,
  View,
} from "react-native";
import { useDispatch, useSelector } from "react-redux";

import { AppState } from "../../store/AppState";
import * as userAction from "../../store/actions/user.action";
import Colors from "../../constants/Colors";
import Loader from "../../components/UI/Loader";
import { UserStackScreenProps } from "../../navigation/types";
import User from "../../types/User";
import { styles } from "./styles";

const UsersScreen = ({ navigation }: UserStackScreenProps<"Users">) => {
  const [loading, setLoading] = useState(true);
  const [input, setInput] = useState("");
  const users = useSelector((state: AppState) => state.user.filteredUsers);
  const dispatch = useDispatch();

  useEffect(() => {
    if (users.length > 0) setLoading(false);
  }, [users]);

  useEffect(() => {
    dispatch(userAction.searchUser(input));
  }, [input]);

  const handleOnUserPress = (user: User) => {
    navigation.navigate("UserDetails", { user });
  };

  const renderUser: ListRenderItem<User> = ({ item: user }) => {
    return (
      <Pressable
        style={styles.userItem}
        onPress={() => handleOnUserPress(user)}
      >
        <Text>{user.name}</Text>
        <Text>{user.locality}</Text>
      </Pressable>
    );
  };

  return (
    <View style={styles.container}>
      <TextInput
        value={input}
        style={styles.searchInput}
        placeholderTextColor={Colors.gray}
        placeholder="Search..."
        onChangeText={setInput}
      />

      <FlatList
        style={styles.usersContainer}
        showsVerticalScrollIndicator={false}
        data={users}
        renderItem={renderUser}
        keyExtractor={(user) => user.id.toString()}
      />

      <Loader visible={loading} />
    </View>
  );
};

export default UsersScreen;
