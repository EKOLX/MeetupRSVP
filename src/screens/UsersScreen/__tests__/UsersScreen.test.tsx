import { ReactTestRenderer, create } from "react-test-renderer";

import UsersScreen from "..";
import { UserStackScreenProps } from "../../../navigation/types";

jest.mock("react-redux", () => {
  const ActualReactRedux = jest.requireActual("react-redux");
  return {
    ...ActualReactRedux,
    useDispatch: jest.fn(),
    useSelector: jest.fn().mockImplementation(() => ({})),
  };
});

describe("Users screen", () => {
  let usersScreen: ReactTestRenderer;

  beforeEach(() => {
    usersScreen = create(
      <UsersScreen {...({} as UserStackScreenProps<"Users">)} />
    );
  });

  it("should match snapshot", () => {
    const tree = usersScreen.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
