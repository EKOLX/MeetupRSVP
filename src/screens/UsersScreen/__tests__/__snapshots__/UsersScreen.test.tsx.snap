// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users screen should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 32,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search..."
    placeholderTextColor="#BACDDB"
    style={
      {
        "borderColor": "#BACDDB",
        "borderRadius": 6,
        "borderWidth": 1,
        "padding": 6,
        "width": "100%",
      }
    }
    value=""
  />
  <RCTScrollView
    data={{}}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    style={
      {
        "marginTop": 22,
        "width": "100%",
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <RCTSafeAreaView
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0.6)",
        "height": 1334,
        "justifyContent": "center",
        "position": "absolute",
        "width": 750,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 9,
          "flexDirection": "row",
          "marginHorizontal": 70,
          "padding": 16,
        }
      }
    >
      <ActivityIndicator
        color="#6D67E4"
        size="large"
      />
      <Text
        style={
          {
            "fontSize": 16,
            "marginLeft": 6,
          }
        }
      >
        Loading...
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
